grants_tagger/utils.py

```diff
@@ -95,3 +68,78 @@ def calc_performance_per_tag(Y_true, Y_pred, tags):
             'f1': f1_score(y_true_tag, y_pred_tag)
         })
     return pd.DataFrame(metrics)
+
+def yield_texts(data_path):
+    with open(data_path) as f:
+        for line in f:
+            item = json.loads(line)
+            yield item["text"]
+
+def yield_tags(data_path):
+    with open(data_path) as f:
+        for line in f:
+            item = json.loads(line)
+            yield item["tags"]
+
+def load_dataset(data_path, tokenizer, label_binarizer, sparse_labels=False, data_cache=None, random_seed=42, 
+                 shuffle=True, shuffle_buffer=1000, load_buffer=1000):
+    def transform_data(texts, tags):
+        text_encoded = tokenizer.transform(texts)
+        tags_encoded = label_binarizer.transform(tags)
+
+        if sparse_labels:
+            tags_encoded = tags_encoded.todense() # returns matrix
+            tags_encoded = np.asarray(tags_encoded)
+        
+        return text_encoded, tags_encoded
+
+    def data_gen():
```

Comment:
it would be nice for some comments on this or a function rename, since it isnt just a data generator but a kind of windowed data generator?

Reply:
I can add comments for sure, guilty on that, need to write more docstrings 😬 

this is actually a callable generator that returns one example at a time. why callable? because when it returns all elements tensorflow needs a way to restart so it basically calls it again, otherwise it would be able to iterate only once.

```
def data_gen():
    for i in range(5):
        yield I

non_callable_generator = data_gen()
callable_generator = data_gen
```